bfvmm/integration/arch/intel_x64/test_all.cpp

```diff
@@ -26,82 +26,30 @@
 //     from the EPT map.
 //
 
-#include <bfcallonce.h>
+#include <vmm.h>
 
-#include <bfvmm/vcpu/vcpu_factory.h>
-#include <bfvmm/hve/arch/intel_x64/vcpu.h>
-
-using namespace bfvmm::intel_x64;
-
-// -----------------------------------------------------------------------------
-// vCPU
-// -----------------------------------------------------------------------------
-
-namespace test
-{
-
-bfn::once_flag flag;
 ept::mmap g_guest_map;
 
-class vcpu : public bfvmm::intel_x64::vcpu
+void
```

Comment:
Notice the removal of the domain object. I do not believe this is needed for the base as it doesn't do anything without guest support. Also, notice that we don't return bool. If you have an error, throw.